PyViCare/PyViCareDeviceConfig.py

```diff
@@ -73,10 +73,13 @@ def asAutoDetectDevice(self):
     def get_raw_json(self):
         return self.service.fetch_all_features()
 
-    def dump_secure(self):
-        dumpJSON = json.dumps(self.get_raw_json(), indent=4)
+    def dump_secure(self, flat=False):
+        dumpJSON = json.dumps(self.get_raw_json(), indent=None if flat else 4)
 
         def repl(m):
             return m.group(1) + ('#' * len(m.group(2))) + m.group(3)
 
-        return re.sub(r'(["\/])(\d{6,})(["\/])', repl, dumpJSON)
+        dumpJSON = re.sub(r'(["\/])(\d{6,})(["\/])', repl, dumpJSON)
+        if not flat:
+            return dumpJSON
+        return re.sub(r'{"apiVersion', '\n{"apiVersion', dumpJSON)
```

Comment:
There is no guarantee that "apiVersion" will be the first element in the object. You are also using a regex replace, in that case a normal string replace would be sufficient.

It would be great to have an alternative solution for adding newlines.